tower/src/builder/mod.rs

```diff
@@ -197,11 +197,23 @@ impl<L> ServiceBuilder<L> {
         f: F,
     ) -> ServiceBuilder<Stack<crate::util::MapRequestLayer<F>, L>>
     where
-        F: Fn(R1) -> R2,
+        F: FnMut(R1) -> R2 + Clone,
```

Comment:
Is the `Clone` actually required here? I'd think it's only required if you want to be able to clone the `MapRequest<S, F>`.

Reply:
The `Clone` bound is required by the `Layer`, since `Layer::layer` takes `&self`. The same layer can be used to wrap any number of services, so the `MapRequestLayer` has to clone the function each time it creates a service. 

Similarly, the same `ServiceBuilder` _could_ actually be used to wrap any number of root services...although I notice that [the current `ServiceBuilder::service` method takes `self` by value](https://github.com/tower-rs/tower/blob/57e8412ae4a31343e78363288e1cd9de4898b229/tower/src/builder/mod.rs#L237-L238). We should maybe change that for 0.4?

Reply:
Oh right, forgot about `Layer` and `&self`. Carry on.